src/mp/proxy.cpp

```diff
@@ -268,6 +267,12 @@ void EventLoop::startAsyncThread(std::unique_lock<std::mutex>& lock)
     }
 }
 
+bool EventLoop::done(std::unique_lock<std::mutex>& lock)
+{
+    assert(m_num_clients >= 0);
+    return m_num_clients == 0 && m_async_fns.empty();
+}
```

Comment:
What is the point of passing `lock` when it is not going to be used? Looks weird. Is this to show that `m_mutex` must be locked when calling this? If yes, then engaging the clang annotations would be more clear. Because now `lock` could be anything, it may not own a mutex or may own any mutex other than `m_mutex`. Maybe add these:

```suggestion
bool EventLoop::done(std::unique_lock<std::mutex>& lock)
{
    assert(m_num_clients >= 0);
    assert(lock.mutex() == &m_mutex);
    assert(lock.owns_lock());
    return m_num_clients == 0 && m_async_fns.empty();
}
```